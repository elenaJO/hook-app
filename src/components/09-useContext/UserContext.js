//es un componente que emarque a sus hijos, para que transfiera su informacion a ellos
import { createContext } from 'react';

// ya creamos el context,es un HOC son compoementes q tiene compoenetes hijos.
export const UserContext = createContext(null);