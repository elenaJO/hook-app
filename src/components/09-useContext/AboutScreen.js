import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const AboutScreen = () => {
	const { user, setUser  } = useContext(UserContext);

	//el context sufrio una modificacion y afecto a los demas
	const handleClick = () => {
		setUser({});
	}

	return (
		<div>
			<h1>AboutScreen</h1>
			<hr/>
			<pre>
				{ JSON.stringify(user, null, 4) }
			</pre>
			<button 
				className="btn btn-warning my-2"
				onClick={ handleClick }
			>
				Logout
			</button>
		</div>
	)
}
