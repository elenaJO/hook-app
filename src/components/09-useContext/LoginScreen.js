import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
	const { setUser} = useContext(UserContext);

	return (
		<div>
			<h1>LoginScreen</h1>
			<hr/>
			<button 
				className="btn btn-primary my-2" 
				onClick={() => setUser({ id:1, name: 'Elena'})}>
				Login
			</button>
		</div>
	)
}
