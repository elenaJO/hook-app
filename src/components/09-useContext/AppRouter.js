//puede tener mas de un router
import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
	Redirect,
} from 'react-router-dom';
import{ AboutScreen } from './AboutScreen';
import{ LoginScreen } from './LoginScreen';
import{ HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
	return (
		<Router>
			{/* recomiendan que este encerrado en un div */}
			<div>
				<NavBar />
				<div className="container">
					<Switch>
						{/* estoy en la ruta /login y me muestra el HomeScreen es por q el path evalua como expresion regular como ve que 
						la ruta tiene / y coincidemuestra el HomeScreen
						<Route path="/" component={ HomeScreen }/> podemos la ruta / abajo o poner exact*/}
						<Route exact path="/" component={ HomeScreen }/>
						<Route exact path="/about" component={ AboutScreen }/>
						<Route exact path="/login" component={ LoginScreen }/>

						{/* //si no encuentra la ruta podemos poner eso ya q es un switch es el default sino ponemos:
						<Route component={ HomeScreen }/> */}
						{/* si no encuentra redirecciona al / */}
						<Redirect to="/"/>
					</Switch>
				</div>
			</div>
		</Router>
	)
}
