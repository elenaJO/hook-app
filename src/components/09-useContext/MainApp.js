//es aplicacion como tal
import React, { useState } from 'react'
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';
export const MainApp = () => {

  const [user, setUser] = useState({});

	return (
    //el UserContext le va a proveer informacion al AppRouter y este a sus hijos, mediante propiedad value
    <UserContext.Provider value={{
      user,
      setUser
    }}>
      <AppRouter />
    </UserContext.Provider>
	)
}
