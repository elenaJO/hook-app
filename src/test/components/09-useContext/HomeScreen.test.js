import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('pruebas en <HomeScreen/>', () => {
	//necesita el user entonces tmb creamos un useContext
	//el shallow solo renderiza en primero componente que es el UserContext, lo unico q hace es rederizar el HomeScreen por eso utilizams el mount
	const user = {
		name: 'Elena',
		email: 'elena@gmail.com'
	};

	const wrapper = mount(
		<UserContext.Provider value={{
			user
		}}>
			<HomeScreen/>
		</UserContext.Provider>
	);
	test('debe mostrarse correctamente', async () => {
		expect(wrapper).toMatchSnapshot();
	})
})