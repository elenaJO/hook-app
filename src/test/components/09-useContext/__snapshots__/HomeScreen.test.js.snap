// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomeScreen/> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
    "name": "Elena",
    "email": "elena@gmail.com"
}
  </pre>
</div>
`;
